### PR TITLE
WIP: Some docker fixes

### PR DESCRIPTION
**Pull request by scottwittenburg**
This still needs a few changes before we can merge it, but it's currently working for me.  Some of the changes this PR makes include:

- make the base container (`bionic-python`) truly Ubuntu 18.04 rather than Debian stretch
- provides options to build the compute/visualize containers based on nvidia docker 
- creates a new EGL build of ParaView
- provides docker-compose override file to allow selection of nvidia containers at runtime

Some things still left to do before merging:

- [x] Clean up commit history
- [x] Rebase on latest master (which brings changes for Girder 3 migration)
- [x] Minor cleanups of container stack
- [x] Update the compute-parflow Dockerfile to allow basing on nvidia stack

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@patrickoleary @aronhelser While this isn't 100% ready to go, it's getting pretty close except for the short TODO list I put in the PR description.

Feel free to take a look or try it out (obviously need an NVidia graphics card with latest drivers in order to try the gpu-accelerate path).

File: docker/celery-pyfr/Dockerfile [hunk 1]
```diff
@@ -6,26 +6,26 @@ LABEL maintainer="patrick.oleary@kitware.com" \
...
-                       openssh-server \
-                       openmpi-bin \
-                       libopenmpi-dev \
-                       libhdf5-dev \
-                       libmetis-dev \
-                       strace \
-                       unzip \
-                       wget \
-                       --no-install-recommends && \
+    apt-get install -y --no-install-recommends \
```
Comment (aronhelser) [inline docker/celery-pyfr/Dockerfile:9]: was dropping build-essential intentional?

Comment (scottwittenburg) [reply docker/celery-pyfr/Dockerfile:9]: I just moved it to the base image (`bionic-python`).

**Comment by scottwittenburg** [comment]
ping @patrickoleary @aronhelser This is ready for you guys to take a look.

**Comment by aronhelser** [comment]
Looks good to me! +1
The only things I noticed is some places doing the 'package cleanup' might have been removed - but I'm not sure if that step is even important?

**Comment by scottwittenburg** [comment]
Thanks for mentioning that, I meant to bring it up with you guys at some point.  I never removed package cleanup, but rather any time it is needed (if `apt-get update` is run and pkgs installed) it all needs to be done within the same `RUN` command.  Doing it at the end still results in the final image not having the cleaned up files, but they'll have to be downloaded with the layer that brought them regardless.  You can read about that [here](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run), near the bottom of the `RUN` section.

**Comment by aronhelser** [comment]
Were you going to merge this? I think we should start using it!

**Comment by scottwittenburg** [comment]
I think we were holding off because @patrickoleary had a demo last week, but we could probably merge it now.  I don't suppose you have tried the branch?  I guess if we merge it, and then it turns out only to work for me, I can revisit in a follow-on topic.

**Comment by patrickoleary** [comment]
I have not tried it, no time right now. Aron can you try it, then merge it.

**Comment by aronhelser** [comment]
I tried it out, and had a problem building some of the paraflow images - Scott is looking at it. I was able to build all the images used by docker-compose, and I did the complete workflow for pyfr through visualizer - so that part is good to go.

**Comment by patrickoleary** [comment]
With NVIDIA?

**Comment by aronhelser** [comment]
Not yet - nvidia doesn't work on windows.  I'm going to try it out on my linux box next, and try it with the nvidia runtime.

**Comment by aronhelser** [comment]
On Linux it works. I started with Docker 19.03, and had to follow [these suggestions](https://github.com/docker/compose/issues/6691#issuecomment-525245786) to enable the 'runtime' keyword in docker-compose. After that, I could run the pyfr workflow all the way through visualization, and the animation was much faster.

**Comment by patrickoleary** [comment]
+1 Add that information to the main README and merge.

**Comment by aronhelser** [comment]
I'll add the readme in a separate commit....

**Pull request merged**